test/tap/lifecycle.js

```diff
@@ -42,3 +42,30 @@ test('lifecycle : rejects wd for ', function (t) {
     t.end()
   })
 })
+
+test('lifecycle : accepts TMPDIR that is writable', function (t) {
+  var wd = '/opt/my-time/node_modules/time'
+
+  lifecycle._getTemporaryDirectory(wd, process.env, function (er, tmpdir) {
+    t.equal(tmpdir, process.env.TMPDIR)
+    t.end()
+  })
+})
+
+test('lifecycle : uses wd if TMPDIR is not writable', function (t) {
+  var wd = process.cwd()
+
+  lifecycle._getTemporaryDirectory(wd, {TMPDIR: '/probably/not/writable'}, function (er, tmpdir) {
+    t.equal(tmpdir, wd)
+    t.end()
+  })
+})
+
+test('lifecycle : errors if neither TMPDIR or wd are writable', function (t) {
+  var wd = '/probably/not/writable/either'
+
+  lifecycle._getTemporaryDirectory(wd, {TMPDIR: '/probably/not/writable'}, function (er, tmpdir) {
+    t.type(er, Error)
```

Comment:
I'm new to tap and `t.error` didn't seem to do what I expected. Is this the proper way to test this?